classes/file/ContextFileManager.php

```diff
@@ -41,7 +41,7 @@ public function __construct($contextId)
     public function getBasePath()
     {
         $dirNames = Application::getFileDirectories();
-        return parent::getBasePath() . $dirNames['context'] . $this->contextId . '/';
+        return rtrim(parent::getBasePath(), '/\\') . $dirNames['context'] . $this->contextId . '/';
```

Comment:
This is more than a syntax change; can you file it separately?

Reply:
Given that the `getBasePath()` is basically a `Config::getVar('files', 'files_dir')`, I imagine that a `files_dir` ending with `/` would lead to double bars `//`, but I didn't test tough.

Reply:
And I think I can recall seeing cases where `//` exists in practice -- OSs tend to be forgiving in this case, which is why it hasn't been fixed yet. But I think this should be kept back for the 3.5.0 cycle.